lib/samples/manage_features/manage_features.dart

```diff
@@ -0,0 +1,433 @@
+            // Display a progress indicator and prevent interaction until state is ready.
+            Visibility(
+              visible: !_ready,
+              child: const SizedBox.expand(
+                child: ColoredBox(
+                  color: Colors.white30,
+                  child: Center(child: CircularProgressIndicator()),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void onMapViewReady() async {
+    // Create and load a service geodatabase from a service URL.
+    const featureServiceURL =
+        'https://sampleserver6.arcgisonline.com/arcgis/rest/services/DamageAssessment/FeatureServer/0';
+    final serviceGeodatabase =
+        ServiceGeodatabase.withUri(Uri.parse(featureServiceURL));
+    await serviceGeodatabase.load();
+
+    // Get the feature table from the service geodatabase referencing the Damage Assessment feature service.
+    // Creating the feature table from the feature service will cause the service geodatabase to be null.
+    final table = serviceGeodatabase.getTable(layerId: 0);
+    if (table != null) {
+      _damageServiceFeatureTable = table;
+      // Load the table.
+      await _damageServiceFeatureTable.load();
+      // Get the required field from the table - in this case, damage type.
+      final damageTypeField =
+          table.fields.firstWhere((field) => field.name == damageTypeFieldName);
+      // Get the domain for the field.
+      final domain = damageTypeField.domain as CodedValueDomain;
+      // Update the dropdown menu with the attribute values from the domain.
+      configureAttributeDropdownMenuItems(domain);
+
+      // Create a feature layer to visualize the features in the table.
+      _damageFeatureLayer =
+          FeatureLayer.withFeatureTable(_damageServiceFeatureTable);
+      // Create a map with the ArcGIS Streets basemap.
+      final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISStreets);
+      // Add the feature layer to the map's operational layers.
+      map.operationalLayers.add(_damageFeatureLayer);
+      // Set the map to the map view controller.
+      _mapViewController.arcGISMap = map;
+      // Zoom to an initial viewpoint.
+      _mapViewController.setViewpoint(
+        Viewpoint.fromCenter(
+          ArcGISPoint(
+            x: -10800000,
+            y: 4500000,
+            spatialReference: SpatialReference.webMercator,
+          ),
+          scale: 3e7,
+        ),
+      );
+      // Set the ready state variable to true to enable the sample UI.
+      setState(() => _ready = true);
+    } else {
+      showMessageDialog('Unable to access the required feature table.');
+    }
+  }
+
+  void onTap(Offset localPosition) async {
+    // Configure actions when a user taps on the map, depending on the selected operation.
+    if (_selectedOperation == FeatureManagementOperation.create) {
+      // Create a feature if create is selected.
+      createFeature(localPosition);
```

Comment:
I think we want `onTap` to `await` on each of the three possibilities (createFeature, updateGeometry, and identifyAndSelectFeature). Which means each of those would have to be declared to return `Future<void>` I think.

Reply:
Ok, I'd removed those as wasn't sure if we were waiting but I wasn't 100% either way on that so I've added back in.